src/Infrastructure/Service/Precondition/NoAbsoluteSymlinksExist.php

```diff
@@ -29,6 +30,10 @@ protected function getDefaultUnfulfilledStatusMessage(): string
 
     protected function isSupportedLink(PathInterface $file, PathInterface $codebaseRootDir): bool
     {
+        if (!$this->filesystem->isSymlink($file)) {
+            return true;
+        }
```

Comment:
☝️ but this is not _ignoring_ hard links, this is making `NoAbsoluteSymlinksExist` react to hard links.

The root cause: 
```
                    if (!$this->filesystem->isLink($file)) {
                        continue;
                    }
```
in `AbstractLinkIteratingPrecondition`.

That means `NoAbsoluteSymlinksExist` is being called for _both_ hard links and symlinks because `isLink()`'s logic says so:
```
    public function isLink(PathInterface $path): bool
    {
        return in_array($this->getFileType($path), [
            self::FILE_IS_HARD_LINK,
            self::FILE_IS_SYMLINK,
        ], true);
    }
```

**Conclusion: either the docs changes in this PR are wrong or `AbstractLinkIteratingPrecondition` is wrong. I think it's the latter. I think `AbstractLinkIteratingPrecondition` should be made aware of whether its concrete subclass cares about only hard links or only symlinks, to then only invoke `isSupportedLink()` for the link type that this concrete subclass is concerned with.**

Reply:
I understand what you're getting at here, but I don't think it's accurate to say that it's _wrong_. Remember that the statement that "Hard links are ignored" is on the _interface_, which makes no promises about implementation details. In terms of the net effect, from the perspective of a client, it not wrong to say that hard links are ignored. Nevertheless, I'll change the language to "It does not apply to hard links" in a follow-up.